packages/nx/src/command-line/release/utils/git.ts

```diff
@@ -44,12 +57,61 @@ function escapeRegExp(string) {
 const SEMVER_REGEX =
   /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/g;
 
+/**
+ * Extract the tag and version from a tag string
+ *
+ * @param tag - The tag string to extract the tag and version from
+ * @param tagRegexp - The regex to use to extract the tag and version from the tag string
+ *
+ * @returns The tag and version
+ */
+export function extractTagAndVersion(
+  tag: string,
+  tagRegexp: string,
+  options: GetLatestGitTagForPatternOptions
+): GitTagAndVersion {
+  const { releaseTagPatternRequireSemver = true } = options;
+
+  const [latestMatchingTag, ...rest] = tag.match(tagRegexp);
+  let version = releaseTagPatternRequireSemver
+    ? rest.filter((r) => {
+        return r.match(SEMVER_REGEX);
+      })[0]
+    : rest[0];
+
+  return {
+    tag: latestMatchingTag,
+    extractedVersion: version ?? null,
+  };
+}
+
+/**
+ * Get the latest git tag for the configured release tag pattern.
+ *
+ * This function will:
+ * - Get all tags from the git repo, sorted by version
+ * - Filter the tags into a list with SEMVER-compliant tags, matching the release tag pattern
+ * - If a preid is provided, prioritise tags for that preid, then semver tags without a preid
+ * - If no preid is provided, search only for stable semver tags (i.e. no pre-release or build metadata)
+ *
+ * @param releaseTagPattern - The pattern to filter the tags list by
+ * @param additionalInterpolationData - Additional data used when interpolating the release tag pattern
+ * @param options - The options to use when getting the latest git tag for the pattern
+ *
+ * @returns The tag and version
+ */
 export async function getLatestGitTagForPattern(
   releaseTagPattern: string,
   additionalInterpolationData = {},
-  checkAllBranchesWhen?: boolean | string[],
-  releaseTagPatternRequireSemver: boolean = true
-): Promise<{ tag: string; extractedVersion: string } | null> {
+  options: GetLatestGitTagForPatternOptions
+): Promise<GitTagAndVersion | null> {
+  const {
+    checkAllBranchesWhen,
+    releaseTagPatternRequireSemver = true,
+    releaseTagPatternStrictPreid,
+    preId = '',
+  } = options;
```

Comment:
This is the same point as above, and I had missed it on a previous PR review, we shouldn't have default values in here